libs/java/instance_provider/src/main/java/com/yahoo/athenz/instance/provider/impl/InstanceK8SProvider.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright The Athenz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.instance.provider.impl;
+
+import com.yahoo.athenz.auth.KeyStore;
+import com.yahoo.athenz.auth.token.IdToken;
+import com.yahoo.athenz.auth.token.jwts.JwtsHelper;
+import com.yahoo.athenz.auth.token.jwts.JwtsSigningKeyResolver;
+import com.yahoo.athenz.common.server.util.config.dynamic.DynamicConfigCsv;
+import com.yahoo.athenz.instance.provider.*;
+import com.yahoo.rdl.JSON;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Header;
+import io.jsonwebtoken.Jwt;
+import io.jsonwebtoken.Jwts;
+import org.eclipse.jetty.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLContext;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import static com.yahoo.athenz.common.server.util.config.ConfigManagerSingleton.CONFIG_MANAGER;
+import static com.yahoo.athenz.instance.provider.impl.InstanceAWSProvider.*;
+import static com.yahoo.athenz.instance.provider.impl.InstanceGCPProvider.*;
+
+public class InstanceK8SProvider implements InstanceProvider {
```

Comment:
so we can't support both aws and gcp at the same time since the provider will only create a single oidcissuervalidator?

Reply:
oh yeah. Great catch! will fix that.

Reply:
actually, we are not creating a single OIDCIssuerValidator but a single OIDCIssuerValidatorFactory which can hand out instance of DefaultAWSOIDCIssuerValidator OR DefaultGCPOIDCIssuerValidator.

Reply:
let's discuss this Monday since I'm not sure this is as generic as we want it to be.